pkg/azuredisk/nodeserver.go

```diff
@@ -222,15 +234,14 @@ func (d *Driver) NodePublishVolume(ctx context.Context, req *csi.NodePublishVolu
 	if req.GetReadonly() {
 		mountOptions = append(mountOptions, "ro")
 	}
-	mountOptions = util.JoinMountOptions(mountFlags, mountOptions)
 
 	klog.V(2).Infof("NodePublishVolume: creating dir %s", target)
 	if err := d.mounter.MakeDir(target); err != nil {
 		return nil, status.Errorf(codes.Internal, "Could not create dir %q: %v", target, err)
 	}
 
 	klog.V(2).Infof("NodePublishVolume: mounting %s at %s", source, target)
-	if err := d.mounter.Mount(source, target, "ext4", mountOptions); err != nil {
+	if err := d.mounter.Mount(source, target, "", mountOptions); err != nil {
```

Comment:
fake the mount and add some unit tests for the options?

Reply:
there are sanity tests for each CSI function and about mount options, it just fetches the options from CSI struct and pass it into standard mounter.Mount func